### PR TITLE
enc, set_hex() behaviour change

### PR DESCRIPTION
**Pull request by EasySec**
slight change in `enc` command **set_hex()** behaviour : constify input, error tolerance and reporting

### CODE REVIEW

File: apps/enc.c [hunk 1]
```diff
@@ -601,22 +601,22 @@ static void show_ciphers(const OBJ_NAME *name, void *arg)
...
     unsigned char j;
 
+    i = size * 2;
     n = strlen(in);
-    if (n > (size * 2)) {
+    if (n > i) {
         BIO_printf(bio_err, "hex string is too long\n");
-        return 0;
-    }
+        n = i; /* ignore exceeding part */
```
Comment (paulidale) [inline apps/enc.c]: I'm not convinced that continuing is a great idea.  At the least, the error message should say thing along the lines of  _hex string is too long, ignoring excess_ so it is completely clear what is happening.

File: apps/enc.c [hunk 2]
```diff
@@ -601,22 +601,22 @@ static void show_ciphers(const OBJ_NAME *name, void *arg)
...
+    } else if (n < i)
+        BIO_printf(bio_err, "hex string is too short\n");
```
Comment (paulidale) [inline apps/enc.c]: Style NIT: this block should be surrounded by '{' & '}' because the previous part of the if is.

Comment (paulidale) [reply apps/enc.c]: You also need a better message explaining what is going to occur.  I.e. _hex string is too short, padding with zero bytes to length_.

File: apps/enc.c (hunk 1 above)
Comment (EasySec) [reply apps/enc.c]: You may be right, but please consider the current situation : an error is raised only when iv is longer than the buffer size, not the **expected** iv length (see change line 464).

Comment (EasySec) [reply apps/enc.c]: I agree with you about the error message. Thanks.

File: apps/enc.c (hunk 2 above)
Comment (EasySec) [reply apps/enc.c]: Agree again. Will be changed, thanks

File: apps/enc.c (hunk 1 above)
Comment (paulidale) [reply apps/enc.c]: What it does now isn't generally a good indication as to what it should do.  Unfortunately.

**Comment by paulidale** [comment]
Merged thanks.

This leaves the code better than before, so the question as to failure or not can be delayed.

File: apps/enc.c [hunk 3]
```diff
@@ -601,22 +601,23 @@ static void show_ciphers(const OBJ_NAME *name, void *arg)
...
+        BIO_printf(bio_err, "hex string is too long, ignoring excess\n");
+        n = i; /* ignore exceeding part */
+    } else if (n < i) {
+        BIO_printf(bio_err, "hex string is too short, padding with zero bytes to length\n");
     }
+
     memset(out, 0, size);
     for (i = 0; i < n; i++) {
-        j = (unsigned char)*in;
-        *(in++) = '\0';
```
Comment (dot-asm) [inline apps/enc.c]: Some would argue that it's appropriate to wipe secret data as soon as you not longer have use for it. This means that they would argue that if set_hex doesn't do it, then caller should.

**Pull request closed without merging**